### DISCUSSION TITLE
Additional php extensions: zip, posix and pcntl

### DISCUSSION THREAD

**Discussion by jkniest** [tiredofit/docker-nginx-php-fpm#25] [Ideas]
Hey there 👋 

Today I tried your docker images for my application and I must say: I LOVE THEM!

Everthing worked flawlessly, - except: three extensions were missing for my application: `zip`, `posix` and `pcntl`. For now I build my own Dockerfile which adds the following lines:

```
# Activate additional extensions
RUN echo "extension=zip.so" >> /etc/php8/conf.d/00-zip.ini && \
    echo "extension=posix.so" >> /etc/php8/conf.d/00-posix.ini && \
    echo "extension=pcntl.so" >> /etc/php8/conf.d/00-pcntl.ini
```

This is already enough to work, so it seams that everything is already installed. I couldn't find the env variables for those. I tried looking through the code to make a PR myself, but tbh I don't understand much of it 👯  But I have no experience with zabbix.

Would that be something that you are willing to implement? Or if you could point my in the right direction, where I can look at the other extensions, I can try to create a PR myself if it helps.

Thanks for your awesome work!

**Comment by tiredofit** [comment]
Thanks for the compliments, I use all these images in production daily and happy to share them with others, I find it very quick to deploy new applications using these base stacks.

Can you tell me with "Variant" (OS+PHP Version) you are seeing this on? I may have messed something up along the way - Those extensions are pretty standard so it's likely an oversight on my end.

**Comment by tiredofit** [comment]
Just to add to this - Here are the default extensions that are active for all images:
````
    PHP_ENABLE_APCU=TRUE \
    PHP_ENABLE_BCMATH=TRUE \
    PHP_ENABLE_BZ2=TRUE \
    PHP_ENABLE_CTYPE=TRUE \
    PHP_ENABLE_CURL=TRUE \
    PHP_ENABLE_DOM=TRUE \
    PHP_ENABLE_EXIF=TRUE \
    PHP_ENABLE_FILEINFO=TRUE \
    PHP_ENABLE_GD=TRUE \
    PHP_ENABLE_ICONV=TRUE \
    PHP_ENABLE_IMAP=TRUE \
    PHP_ENABLE_INTL=TRUE \
    PHP_ENABLE_MBSTRING=TRUE \
    PHP_ENABLE_MYSQLI=TRUE \
    PHP_ENABLE_MYSQLND=TRUE \
    PHP_ENABLE_OPCACHE=TRUE \
    PHP_ENABLE_OPENSSL=TRUE \
    PHP_ENABLE_PDO=TRUE \
    PHP_ENABLE_PDO_MYSQL=TRUE \
    PHP_ENABLE_PGSQL=TRUE \
    PHP_ENABLE_PHAR=TRUE \
    PHP_ENABLE_SESSION=TRUE \
    PHP_ENABLE_SIMPLEXML=TRUE \
    PHP_ENABLE_TOKENIZER=TRUE \
    PHP_ENABLE_XML=TRUE \
    PHP_ENABLE_XMLREADER=TRUE \
    PHP_ENABLE_XMLWRITER=TRUE
````
You theoretically should be able to just add an environment variable when starting the docker image like so: `docker run -e PHP_ENABLE_ZIP=TRUE -e PHP_ENABLE_PCNTL=TRUE -e PHP_ENABLE_POSIX=TRUE tiredofit/nginx-php-fpm`  That should switch them on for you.

**Reply by jkniest** [reply]
Ohhh! I thought they were the only one available! If I can just turn them on using environment variables, that would totally work for me!

I am using a Macbook with the M1 Pro Chip, so arm64 and the image `tiredofit/nginx-php-fpm:alpine-8.0`

Thanks!

**Reply by tiredofit** [reply]
Perfect. I try to make as many things as possible configurable via environment variables. 

If you enter in the container you can get a list of what extensions are available by typing `php-ext list all` and then you can get away with just creating a new environment variable `PHP_ENABLE_(whatever extension)=TRUE` and you should be all set!
